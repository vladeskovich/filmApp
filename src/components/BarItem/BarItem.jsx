import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import Input from '../Input';

const BarItem = ({
  onChange,
  onClick,
  id,
  activeItem,
  name,
}) => {
  const checkedItem = activeItem === id;

  const handleClick = useCallback((event) => {
    const { currentTarget: { dataset: { id } } } = event;

    onClick(id - 1);
  }, [onClick]);

  return (
    <>
      <Input
        checked={checkedItem}
        onChange={onChange}
        type='radio'
        name={name}
        id={name}
      />
      <label
        onClick={handleClick}
        data-id={id}
        htmlFor={name}
      >
      </label>
    </>
  );
};

BarItem.propTypes = {
  onClick: PropTypes.func,
  onChange: PropTypes.func,
  id: PropTypes.number,
  activeItem: PropTypes.number,
  name: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

BarItem.defaultProps = {
  onChange: () => {},
};

export default BarItem;
