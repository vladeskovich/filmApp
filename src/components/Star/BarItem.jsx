import React from 'react';
import PropTypes from 'prop-types';
import Input from '../Input';

const BarItem = ({
  id,
  checked,
  name,
}) => {
  const checkedItem = checked === id;
  const nameId = name + id;
  return (
    <>
      <Input
        checked={checkedItem}
        type='radio'
        name={name}
        id={nameId}
      />
      <label for={nameId}></label>
    </>
  );
};

BarItem.propTypes = {
  checked: PropTypes.bool,
  name: PropTypes.string,
};

BarItem.defaultProps = {};

export default BarItem;
