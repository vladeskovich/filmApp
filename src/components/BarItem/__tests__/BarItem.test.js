import React from 'react';
import BarItem from '../BarItem';

describe('BarItem', () => {
  const props = {
    id: 3,
    onClick: () => {},
    onChange: () => {},
    activeItem: 4,
    name: 'slider-navigation',
  };

  it('should render BarItem component', () => {
    const component = shallow(<BarItem {...props}/>);

    expect(component).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const mockCallBack = jest.fn();
    const component = shallow(<BarItem {...props} activeItem={3} onClick={mockCallBack}/>);

    expect(mockCallBack.mock.calls.length).toBe(0);
    component.find('label').simulate('click', { currentTarget: { dataset: { id: 1 } } });
    expect(mockCallBack.mock.calls.length).toBe(1);
  });
});
