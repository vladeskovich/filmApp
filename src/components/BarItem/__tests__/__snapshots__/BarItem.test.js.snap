// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarItem should render BarItem component 1`] = `
<Fragment>
  <Input
    checked={false}
    id="slider-navigation"
    name="slider-navigation"
    onChange={[Function]}
    type="radio"
  />
  <label
    data-id={3}
    htmlFor="slider-navigation"
    onClick={[Function]}
  />
</Fragment>
`;
